activesupport/lib/active_support/testing/assertionless_tests.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module Testing
+    # Allows to configure the behavior when the test case has no assertions in it.
+    # This is helpful in detecting broken tests that do not perform intended assertions.
+    module AssertionlessTests # :nodoc:
+      def self.prepended(klass)
+        klass.singleton_class.attr_reader :assertionless_tests_behavior
+        klass.extend ClassMethods
+        klass.assertionless_tests_behavior = :ignore
+      end
+
+      module ClassMethods
+        def assertionless_tests_behavior=(behavior)
+          @assertionless_tests_behavior =
+            case behavior
+            when :ignore
+              nil
+            when :log
+              logger =
+                if defined?(Rails.logger) && Rails.logger
+                  Rails.logger
+                else
+                  require "active_support/logger"
+                  ActiveSupport::Logger.new($stderr)
+                end
+
+              ->(message) { logger.warn(message) }
+            when :raise
+              ->(message) { raise Minitest::Assertion, message }
+            else
+              raise ArgumentError, "assertionless_tests_behavior must be one of :ignore, :log, or :raise."
```

Comment:
Probably `YAGNI`, but we could accept any callable here.

Reply:
Added.
I noticed, that `.with` helper is working only with objects, and not with modules. So something like
```ruby
ActiveSupport::TestCase.with(assertionless_tests_behavior: :ignore, &block)
```
is not working. Was that intentional?

Reply:
I just confirmed it does, I suspect that core_ext just instead loaded?